var/spack/repos/builtin/packages/strumpack/package.py

```diff
@@ -67,7 +67,7 @@ class Strumpack(CMakePackage, CudaPackage, ROCmPackage):
     depends_on("mpi", when="+mpi")
     depends_on("blas")
     depends_on("lapack")
-    depends_on("openblas threads=openmp", when="^openblas")
+    depends_on("openblas threads=openmp", when="^[virtuals=lapack,blas] openblas")
```

Comment:
I don't follow from the concretizer changes why this is necessary, and it concerns me a bit. Does it matter which virtuals openblas is providing whether this requirement holds?

This brings to mind a related worry -- if `foo` depends unconditionally on `openblas` (not through the virtual interface, and `bar` depends on `foo` and `blas`, does this implementation prevent `bar ^[virtuals=blas] intel-parallel-studio`? I would think it should (since `foo` is linking `libopenblas.so`).

Reply:
> I don't follow from the concretizer changes why this is necessary, and it concerns me a bit. Does it matter which virtuals openblas is providing whether this requirement holds?

It is not necessary, but having or not the virtuals has an effect. I added a proper test in https://github.com/spack/spack/pull/35322/commits/bfce9f8add86c667f34ef2533a12aa77e775626a to check that virtuals on edges are accounted for in conditions.

In any case, I can revert this specific change (probably this went in accidentally while I was developing).

Reply:
Reverted in https://github.com/spack/spack/pull/35322/commits/56c362f65b397b64c3b98b50548806a963a8d622

This was to test that an edge attribute matters if part of a condition, but there is also a unit test to ensure this semantic works as intended.

Reply:
To be clear, this would matter in cases where we need to specialize on a dependency that could provide more than one virtual. For instance:
```python
depends_on("foo+intel", when="^intel-parallel-studio")
```
would trigger the dependency on `foo+bar` if `intel-parallel-studio` is a dependency too. While:
```python
depends_on("foo+intel", when="^[virtuals=mpi] intel-parallel-studio")
```
is more specific and would trigger the dependency only if `intel-parallel-studio` is the `mpi` provider.